pkg/registry/core/next/registry_client.go

```diff
@@ -49,26 +49,30 @@ func NewWrappedRegistryClient(wrapper RegistryClientWrapper, clients ...registry
 
 // NewRegistryClient - chains together clients into a single registry.NetworkServiceRegistryClient
 func NewRegistryClient(clients []registry.NetworkServiceRegistryClient) registry.NetworkServiceRegistryClient {
-	return NewWrappedRegistryClient(nil, clients...)
+	return NewWrappedRegistryClient(notWrapClient, clients...)
 }
 
 func (n *nextRegistryClient) RegisterNSE(ctx context.Context, request *registry.NSERegistration, opts ...grpc.CallOption) (*registry.NSERegistration, error) {
 	if n.index+1 < len(n.clients) {
 		return n.clients[n.index].RegisterNSE(withNextRegistryClient(ctx, &nextRegistryClient{clients: n.clients, index: n.index + 1}), request)
 	}
-	return n.clients[n.index].RegisterNSE(withNextRegistryClient(ctx, nil), request, opts...)
+	return n.clients[n.index].RegisterNSE(withNextRegistryClient(ctx, &tailRegistryClient{}), request, opts...)
 }
 
 func (n *nextRegistryClient) BulkRegisterNSE(ctx context.Context, opts ...grpc.CallOption) (registry.NetworkServiceRegistry_BulkRegisterNSEClient, error) {
 	if n.index+1 < len(n.clients) {
 		return n.clients[n.index].BulkRegisterNSE(withNextRegistryClient(ctx, &nextRegistryClient{clients: n.clients, index: n.index + 1}), opts...)
 	}
-	return n.clients[n.index].BulkRegisterNSE(withNextRegistryClient(ctx, nil), opts...)
+	return n.clients[n.index].BulkRegisterNSE(withNextRegistryClient(ctx, &tailRegistryClient{}), opts...)
```

Comment:
Move the &tailRegistryClient{} to context.go (see https://github.com/networkservicemesh/sdk/pull/106 for the example for chaining networkservice).

Reply:
This makes it safe to run a chain element even its not in a chain.

Reply:
Ok, did it